.github/workflows/preview.yaml

```diff
@@ -0,0 +1,48 @@
+name: PR Preview
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Probably no need to fetch the entire log, I'd remove it, by default it's `fetch-depth: 1`.